informer/informer.go

```diff
@@ -98,6 +103,19 @@ func New(config Config) (*Informer, error) {
 	return newInformer, nil
 }
 
+func (i *Informer) Boot(ctx context.Context) error {
+	err := prometheus.Register(prometheus.Collector(i))
+
+	if _, ok := err.(prometheus.AlreadyRegisteredError); ok {
```

Comment:
We usually want to align with our error handling conventions. See for instance https://github.com/giantswarm/aws-operator/blob/b322a4d5cc1c7202742b544f546972ae26a5e2af/service/controller/v10/resource/s3bucket/error.go#L23-L44. And while we are at it, lets document this in https://github.com/giantswarm/fmt/blob/master/go/general_style_guide.md. The reason for this is to redirect the error matching magic to a certain place and "authority" to keep the actual code clean and understandable.

Reply:
I changed this, PTAL.
Working on PR for fmt.